scripts/output/comparison/checkClimatePercentiles.R

```diff
@@ -0,0 +1,105 @@
+# of MAGICC at a time, picked in cfg$climate_assessment_magicc_prob_file_iteration
+# The available configuration files were designed to roughly reproduce a certain quantile outcome
+# of 2100 temperature for scenarios that are as similar as possible to an RCP (hence the nomenclature
+# of the config files e.g. RCP20_50.json will be close to the 50th percentile outcome in a RCP2.0-like
+# scenario)
+# This quantile outcome is of course not guaranteed, so this script helps check the validity of
+# the assumed quantile against the 600-member MAGICC7 AR6 assessment that is run by default
+# at the end of each run. Therefore, for it to work, the output script MAGICC7_AR6.R must
+# have already run and added the `MAGICC7 AR6|**` variables to `REMIND_generic_*.mif`.
+#
+# To verify the output, check the figures in `check_output_figures.pdf` and the table in
+# `check_output_summary.csv`
+
+require(tidyverse)
+require(quitte)
+require(gdx)
+require(yaml)
+
+cat("===================== STARTING checkClimatePercentiles.R", "\n")
+cat("===================== see output for final location of output files", "\n")
+lucode2::readArgs("outputdirs")
+runfolders <- outputdirs
+
+checkscenarios <- basename(runfolders)
+cat("checkscenarios", checkscenarios, "\n")
+
+
+# Get a value from the cfg file in a folder
+get_cfgval <- function(runfolder, name) {
+    read_yaml(paste0(runfolder,"/cfg.txt"))[[name]]
+}
+
+# Get the configuration names to have as a reference
+usedconfigs <- bind_rows(lapply(1:length(runfolders),
+    # \(i) read_yaml(paste0(runfolders,"/cfg.txt")[i])$climate_assessment_magicc_prob_file_iteration %>% basename %>% str_remove(".json$")
+    \(i) tibble(
+        scenario = get_cfgval(runfolders[i], "title"), 
+        usedconfig = get_cfgval(runfolders[i], "climate_assessment_magicc_prob_file_iteration") %>% basename %>% str_remove(".json$")
+        )
+))
+
+# Read pm_globalMeanTemperature from the MAGICC gdxs
+tempgdxs <- paste0(runfolders,"/p15_magicc_temp.gdx")
+gdxdata <- bind_rows(
+    lapply(
+    1:length(checkscenarios),
+    \(i) mutate(read.gdx(tempgdxs[i], "pm_globalMeanTemperature"),scenario = get_cfgval(dirname(tempgdxs[i]), "title")
+    ))) %>%
+    rename(period = tall) %>%
+    mutate(variable = "pm_globalMeanTemperature", region = "World", unit = "K") %>%
+    mutate(model = "MAGICC7 In-GAMS")
+
+
+
+# Read reporting data. Requires MAGICC7_AR6 to have been successfully run
+filelist <- list.files(runfolders, pattern = ".mif", full.names = TRUE) 
+miflist <- filelist[str_detect(filelist, "REMIND_generic_.*")&!str_detect(filelist, "withoutPlus")]
+
+mifdata <- read.quitte(miflist)
+miftempdata <- mifdata %>% 
+    filter(region == "World", variable %in% c(
+        "MAGICC7 AR6|Surface Temperature (GSAT)|10.0th Percentile",
+        "MAGICC7 AR6|Surface Temperature (GSAT)|33.0th Percentile",
+        "MAGICC7 AR6|Surface Temperature (GSAT)|50.0th Percentile",
+        "MAGICC7 AR6|Surface Temperature (GSAT)|66.0th Percentile",
```

Comment:
Maybe a minor issue, but why are you using 66.0th Percentile? At least in NGFS, we always use 33.0th and 67.0th.

Reply:
Good point, that was an oversight